grails-app/services/org/pih/warehouse/invoice/InvoiceService.groovy

```diff
@@ -460,4 +468,48 @@ class InvoiceService {
         }
         return csv
     }
+
+    CSVPrinter getInvoiceItemsCsv(List<InvoiceItem> invoiceItems) {
+        CSVPrinter csv = CSVUtils.getCSVPrinter()
+        csv.printRecord(
+                "Invoice Number",
+                "Vendor Invoice Number",
+                "Vendor",
+                "Currency",
+                "Status",
+                "Buyer Organization",
+                "Invoice Type",
+                "Code",
+                "Name",
+                "Order Number",
+                "GL Account",
+                "Budget Code",
+                "Quantity",
+                "Quantity per UoM",
+                "Amount",
+                "Total Amount",
+        )
+
+        invoiceItems?.each { InvoiceItem invoiceItem ->
+            csv.printRecord(
+                    invoiceItem.invoice?.invoiceNumber,
+                    invoiceItem.invoice?.vendorInvoiceNumber,
+                    "${invoiceItem.invoice?.party?.code} ${invoiceItem.invoice?.party?.name}",
+                    invoiceItem.invoice?.currencyUom.name,
+                    invoiceItem.invoice?.status?.name(),
+                    "${invoiceItem.invoice?.partyFrom?.code} ${invoiceItem.invoice?.partyFrom?.name}",
+                    invoiceItem.invoice?.invoiceType?.code?.name() ?: "INVOICE",
+                    invoiceItem.product?.productCode ?: "all",
```

Comment:
what does "all" mean here?

Reply:
on invoice show page we have a similar case where we label adjustments as all since adjustments don't have a product code
https://github.com/openboxes/openboxes/blob/3627a1dd5fcf53edcb7f497ff5c07680b0d1e446/grails-app/views/invoice/_invoiceItems.gsp#L23-L25